repoclosure

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+DNF_VAR_foremanver=nightly
+DNF_VAR_katellover=nightly
+DNF_VAR_candlepinver=nightly
+DNF_VAR_puppetver=7
+
+export DNF_VAR_foremanver DNF_VAR_katellover DNF_VAR_candlepinver DNF_VAR_puppetver
+
+repoclosure() {
+	local distro releasever module
+	distro=./distros/$1
+	releasever=$2
+	module=$3
+
+	if [[ -n $module ]] ; then
+		# DNF needs root to enable modules
+		INSTALLROOT=$(mktemp -d)
+
+		fakeroot ./dnfwrapper "$distro" "$releasever" module enable "$module" -y --installroot "$INSTALLROOT"
+		fakeroot ./dnfwrapper "$distro" "$releasever" repoclosure --installroot "$INSTALLROOT" "${@:4}"
+
+		rm -rf "$INSTALLROOT"
+	else
+		./dnfwrapper "$distro" "$releasever" repoclosure "${@:4}"
+	fi
+}
+
+repoclosure centos-stream 9 "" --check foreman-${DNF_VAR_foremanver}-staging --enablerepo puppet7
+repoclosure centos-stream-legacy 8 foreman-devel:el8 --check foreman-${DNF_VAR_foremanver}-staging --enablerepo puppet7
```

Comment:
this hard-codes closure for foreman.
how would I call it for katello?
how for the case where we add a repo on the CLI to perform closure on a copr build?

Reply:
This is the area where I ended the day of learning. I imagine the next step would be to something work this into obal so we can use it for our `*-packaging` repositories. Ideally set the variables from `package_manifest.yaml`.

Still, I think this script gives a good start to see the general idea.